internal/core/capi.h

```diff
@@ -123,19 +123,19 @@ int fence_metadata(
 int horizon_offset(
     Context* ctx,
     DataHandle* handle,
-    RegularSurface* surface,
-    float above,
-    float below,
+    RegularSurface* primary,
+    RegularSurface* top,
+    RegularSurface* bottom,
     size_t* out,
     size_t out_size
 );
 
 int horizon(
     Context* ctx,
     DataHandle* handle,
-    RegularSurface* surface,
-    float above,
-    float below,
+    RegularSurface* primary,
+    RegularSurface* top,
+    RegularSurface* bottom,
```

Comment:
>In general, primary is still the same "data" which we had in previous horizons. Your reference points.

Why isn't it called reference then?

>And I like passing the same 3 surfaces around in all 3 calls :) Seems clear to me.

You just said that one of them changes definition based the context in which the original request was formed. A context which is lost at this point. Opposite of clear 😄 

> Remember that when we are interpolating we need to pass down the reference point - the one we anchor data with. And for "nearest" calculations we also need a reference point.


Sure, but that doesn't need to be the middle/reference point. It might as well be the top point.

> Not exactly sure we want to do it, but it wasn't part of the task anyway.

If you set out on a task to unify two concept, then I would argue it _is_ a part of the task to make abstraction that makes sense for the new unified codebase. Not just jamming new stuff into old concepts. (VerticalWindow was terrible before, but in this context it's even worse)

Reply:
Trust me, I'm not all all attached to VerticalWindow 😂 My point is that now that we have surfaces for above/below I _think_ we can solve the task that VerticalWindow solves, using only these surfaces. But we leave it for the future.